### PR TITLE
feat: Add $ifNotExists tot $set operator

### PR DESCRIPTION
**Pull request by onhate**
Allow to set a value only if it does not exists on the record, a good use case is on upsert and setting a `createdAt` property

```js
{ $set:  { createdAt: { $ifNotExists: date() } } }
```

https://docs.amazonaws.cn/en_us/amazondynamodb/latest/developerguide/Expressions.UpdateExpressions.html#Expressions.UpdateExpressions.SET.PreventingAttributeOverwrites

extra: allow undefined/null where clause (to be ignored)

### CODE REVIEW

**Comment by onhate** [comment]
@SamVerschueren what do you think about this one?

**Comment by onhate** [comment]
@SamVerschueren ?

**Comment by onhate** [comment]
@SamVerschueren  ?

File: src/lib/types/update-query.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export interface UpdateQuery {
-	$set?: { [key: string]: any };
+	$set?: { [key: string]: { $ifNotExists: any } | any };
```
Comment (SamVerschueren) [inline src/lib/types/update-query.ts:2]: Ideally we expand this interface with all possible operators. But that feels like a future improvement 👍 .

**Review by SamVerschueren — APPROVED** [review]
Awesome job @onhate!

**Pull request merged**